src/View/Requirements_Backend.php

```diff
@@ -1421,6 +1439,39 @@ function () use ($fileList, $minify, $type) {
         return $combinedURL;
     }
 
+    /**
+     * Resolves relative paths in CSS files which are lost when combining them
+     *
+     * @param string $content
+     * @param string $filePath
+     * @return string New content with paths resolved
+     */
+    protected function resolveCSSReferences($content, $filePath)
+    {
+        $doResolving = Config::inst()->get(__CLASS__, 'resolve_relative_css_refs');
+        if (!$doResolving) {
+            return $content;
+        }
+        $fileUrl = Injector::inst()->get(ResourceURLGenerator::class)->urlForResource($filePath);
+        $fileUrlDir = dirname($fileUrl);
+        $content = preg_replace_callback('#([\s\'"\(])((:?[\.]{1,2}/)+)#', function ($a) use ($fileUrlDir) {
+            [ $_, $prefix, $match ] = $a;
+            $full = $fileUrlDir . '/' . $match;
+            $full = preg_replace('#/{2,}#', '/', $full); // ensure there's no repeated slashes
+            while (strpos($full, './') > 0) {
+                $post = $full;
+                $post = preg_replace('#([^/\.]+)/\.\./#', '', $post); // erase 'something/../' with the predecessor
+                $post = preg_replace('#([^/\.]+)/\./#', '\\1/', $post); // erase './'
+                if ($post == $full) {
+                    break; // nothing changed
+                }
+                $full = $post;
+            }
```

Comment:
We may as well use a pre-existing implementation for this instead.
```suggestion
            $full = Path::join(Path::makeRelative($full, $fileUrlDir), $fileUrlDir);
```
Where `Path` here is `Symfony\Component\Filesystem\Path` - you'll need to add that `use` statement at the top of the file.

Reply:
I tried that, but it produces incorrect results, because it refuses to treat consecutive "../".
see `vendor/symfony/filesystem/Path.php:739`: where it says `Don't collapse ".." if the previous part is also ".."`.

Symfony's `Path` has algorithms are based on arrays instead of regular expressions. It may worth building the discussed piece with arrays for better performance (regexp is heavy, right?)... but I don't know where such method would be appropriate - alongsize in `Requirements_Backend` as a private method or maybe in `SilverStripe\Core\Path`?